actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -248,5 +250,9 @@ def _ensure_url_is_http_header_safe(url)
           raise UnsafeRedirectError, msg
         end
       end
+
+      def _allowed_redirect_hosts_permissions
+        @allowed_redirect_hosts_permissions ||= ActionDispatch::HostAuthorization::Permissions.new(allowed_redirect_hosts)
```

Comment:
`ActionDispatch::HostAuthorization::Permissions.new` compile regexps, hence we shouldn't create one per request.